docs/adr/03_egress_filtering.md

```diff
@@ -0,0 +1,58 @@
+# Egress filtering
+
+## Aims
+
+We will filter outbound traffic from the VPC in order to mitigate certain classes of vulnerabilities and limit data exfiltration.
+This will need to be flexible enough to allow the existing applications to install and run with minimal modification.
+
+Note this only applies to connections initiated from within the VPC's private subnets, public subnets and return traffic via load balancers will not be affected.
+
+## Existing infrastructure
+
+The existing DataMart infrastructure filters egress traffic using Squid proxy.
+
+## Constraints
+
+We need to be able to filter HTTP and TLS traffic by domain. IP and port allowlisting will not be sufficient.
+
+Our resources will need to be able to access S3 and other AWS services.
+
+## Design
+
+### AWS services
+
+Allowing unrestricted access to S3 would undermine the rest of the filtering to a large extent, as buckets could be attacker controlled.
+We will instead use VPC endpoints for S3 (and other AWS services) with policies to control what buckets can be accessed.
+
+### AWS Network Firewall
+
+We investigated AWS Network Firewall as a managed alternative to Squid proxy.
+
+Advantages of AWS Network Firewall:
+
+* No server to manage and patch
+* Native integration with AWS CloudWatch
+* Can easily scale to multiple endpoints for redundancy and large amounts of traffic
+  * We do not expect to need either so are largely disregarding this
+
```

Comment:
The native integration is with more of AWS than just CloudWatch - you can mention that there's potential future advantages like sharing policies/monitoring across different DLUHC accounts.